### ISSUE TITLE
Breaking change in 0.1.17?

### ISSUE THREAD

**Issue by ClementParis016** [apollostack/graphql-tag#19]
I was using graphql-tag 0.1.16 along apollo-client 0.5.0, then updated to 0.1.17 and 0.5.6 respectively, which introduced the following `TypeError: Cannot read property 'kind' of undefined` raised from graphql-tag on GraphQL mutations.

This is what my mutation looks like:
```js
  apollo.mutate({
      mutation: gql`
        mutation {
          newEntry(input: {
            name: "${entry.name}",
            description: "${entry.description}"
          }) {
            entryInfo {
              name,
              description
            }
          }
        }
      `
    });
```

Though I'm really new to GraphQL so I'm afraid I can't get what `kind` could refer to :/

**Comment by stubailo** [comment]
@tmeasday perhaps we should have a check for the data being a string so that interpolation keeps working?

@ClementParis016 the right way to do this is via variables. Using string interpolation for dynamic inputs is not good because you could end up with escaping issues.

**Comment by tmeasday** [comment]
Oh, ughh, this is a bad bug. We should have a test for this. Even if we don't encourage it, people defintitely do it.

**Comment by stubailo** [comment]
Agreed.

**Comment by tmeasday** [comment]
@ClementParis016  for the record: I think you were interpolating `undefined`, so there is probably a bug at your end --  I doubt that `name: "undefined"` was what you wanted in your query! 

In any case, this is fixed in 1.1.2

**Comment by ClementParis016** [comment]
Thanks for your help. I sort of felt that using interpolation wasn't great because I had to wrap strings in quotes to get it to work.
I will look into using GraphQL variables instead, thank you 👍
